examples/video.py

```diff
@@ -1,193 +1,276 @@
+    def current_frame(self) -> int:
+        return self._current_frame
+
+    @property
+    def target_frame(self) -> int:
+        return self._target_frame
 
-    def next_frame(self) -> av.plane.Plane:
-        """Get RGB data for the next frame.
-        A VideoPlane is returned containing the RGB data.
-        This objects supports the buffer protocol and can be written to a texture directly.
-        """
-        return next(self._frames)
+    @property
+    def is_paused(self) -> bool:
+        return self._paused
 
 
-class VideoTest(moderngl_window.WindowConfig):
+class VideoPlayerWindow(moderngl_window.WindowConfig):
+    """ModernGL window configuration for video playback."""
+
     gl_version = (3, 3)
     title = "Video Player"
     resource_dir = Path(__file__).parent.resolve() / "resources"
+    vsync = True
+    seek_time = 1.0  # Seconds to seek when using arrow keys
 
     def __init__(self, **kwargs):
         super().__init__(**kwargs)
 
-        self.player = Player(self.ctx, self.resource_dir / "videos/Lightning - 33049.mp4")
-        print("duration   :", self.player.duration)
-        print("fps        :", self.player.fps)
-        print("video_size :", self.player.video_size)
-        print("frames     :", self.player.frames)
-        print("step       :", self.player._decoder.frame_step)
+        # Initialize video player
+        video_path = self.resource_dir / "videos" / "Lightning - 33049.mp4"
+
+        self.player = VideoPlayer(self.ctx, video_path)
 
+        # Setup rendering
         self.quad = geometry.quad_fs()
         self.program = self.load_program("programs/texture_flipped.glsl")
 
-    def on_render(self, time: float, frametime: float):
-        self.player.update(math.fmod(time, 5))
+        # Setup stats printing
+        self._last_print_time = 0
+
+    def on_render(self, time: float, frametime: float) -> None:
+        """Render frame."""
+        if self.player.update(time):  # Check if video ended
+            self.timer.time = 0  # Reset timer if video ended
+
+        # Render video
         self.player.texture.use(0)
         self.quad.render(self.program)
 
-    def on_key_event(self, key, action, modifiers):
+        # Print debug stats every 0.5 seconds regardless of pause state
+        if (time - self._last_print_time) >= 0.5 or time < self._last_print_time:
+            # Get FPS values with safety checks
+            fps_avg = self.timer.fps_average if self.timer.time > 0 else 0.0
+
+            print(
+                f"\rMovie Target FPS: {self.player.fps:.1f} | "
+                f"Window FPS: {fps_avg:.1f} | "
+                f"Frame: {self.player.current_frame}/{self.player.frames} | "
+                f"Time: {self.timer.time:.2f}/{self.player.duration:.2f} | "
+                f"Frame Diff: {self.player.target_frame - self.player.current_frame} | "
+                f"Paused: {self.player.is_paused}",
+                end="",
+                flush=True,
+            )
```

Comment:
maybe using logging via `logger` here (`debug` infos for example) instead of raw `print` ... ?

Reply:
It's an example so it might not be that big of a deal. Still doesn't hurt.